esp_serial_slave_link/essl_sdio.c

```diff
@@ -0,0 +1,496 @@
+    err = essl_sdio_read_bytes(ctx->card, HOST_SLCHOST_PKT_LEN_REG, (uint8_t *) &len, 4);
+    if (err != ESP_OK) {
+        return err;
+    }
+    len &= RX_BYTE_MASK;
+    ctx->rx_got_bytes_latest = len;
+    return ESP_OK;
+}
+
+
+esp_err_t essl_sdio_write_reg(void *arg, uint8_t addr, uint8_t value, uint8_t *value_o, uint32_t wait_ms)
+{
+    ESP_LOGV(TAG, "write_reg: %08X", value);
+    // addrress over range
+    if (addr >= 60) {
+        return ESP_ERR_INVALID_ARG;
+    }
+    //W7 is reserved for interrupts
+    if (addr >= 28) {
+        addr += 4;
+    }
+    return essl_sdio_write_byte(((essl_sdio_context_t *)arg)->card, HOST_SLCHOST_CONF_W_REG(addr), value, value_o);
+}
+
+esp_err_t essl_sdio_read_reg(void *arg, uint8_t add, uint8_t *value_o, uint32_t wait_ms)
+{
+    ESP_LOGV(TAG, "read_reg");
+    // address over range
+    if (add >= 60) {
+        return ESP_ERR_INVALID_ARG;
+    }
+    //W7 is reserved for interrupts
+    if (add >= 28) {
+        add += 4;
+    }
+    esp_err_t ret = essl_sdio_read_byte(((essl_sdio_context_t *)arg)->card, HOST_SLCHOST_CONF_W_REG(add), value_o);
+    ESP_LOGV(TAG, "reg: %08X", *value_o);
+    return ret;
+}
+
+esp_err_t essl_sdio_clear_intr(void *arg, uint32_t intr_mask, uint32_t wait_ms)
+{
+    ESP_LOGV(TAG, "clear_intr: %08X", intr_mask);
+    return essl_sdio_write_bytes(((essl_sdio_context_t *) arg)->card, HOST_SLC0HOST_INT_CLR_REG, (uint8_t *) &intr_mask, 4);
+}
+
+esp_err_t essl_sdio_get_intr(void *arg, uint32_t *intr_raw, uint32_t *intr_st, uint32_t wait_ms)
+{
+    essl_sdio_context_t *ctx = arg;
+    esp_err_t r;
+    ESP_LOGV(TAG, "get_intr");
+    if (intr_raw == NULL && intr_st == NULL) {
+        return ESP_ERR_INVALID_ARG;
+    }
+
+    if (intr_raw != NULL) {
+        r = essl_sdio_read_bytes(ctx->card, HOST_SLC0HOST_INT_RAW_REG, (uint8_t *) intr_raw, 4);
+        if (r != ESP_OK) {
+            return r;
+        }
+    }
+    if (intr_st != NULL) {
+        r = essl_sdio_read_bytes(ctx->card, HOST_SLC0HOST_INT_ST_REG, (uint8_t *) intr_st, 4);
+        if (r != ESP_OK) {
+            return r;
+        }
+    }
+    return ESP_OK;
+}
+
+esp_err_t essl_sdio_set_intr_ena(void *arg, uint32_t ena_mask, uint32_t wait_ms)
+{
+    ESP_LOGV(TAG, "set_intr_ena: %08X", ena_mask);
+    return essl_sdio_write_bytes(((essl_sdio_context_t *)arg)->card, HOST_SLC0HOST_FUNC1_INT_ENA_REG,
+                                 (uint8_t *) &ena_mask, 4);
+}
+
+esp_err_t essl_sdio_get_intr_ena(void *arg, uint32_t *ena_mask_o, uint32_t wait_ms)
+{
+    ESP_LOGV(TAG, "get_intr_ena");
+    esp_err_t ret = essl_sdio_read_bytes(((essl_sdio_context_t *)arg)->card, HOST_SLC0HOST_FUNC1_INT_ENA_REG,
+                                         (uint8_t *) ena_mask_o, 4);
+    ESP_LOGV(TAG, "ena: %08X", *ena_mask_o);
+    return ret;
+}
+
+esp_err_t essl_sdio_send_slave_intr(void *arg, uint32_t intr_mask, uint32_t wait_ms)
+{
+    ESP_LOGV(TAG, "send_slave_intr: %02x", intr_mask);
```

Comment:
You can solve your build errors by using macros from `inttypes.h`
```suggestion
    ESP_LOGV(TAG, "send_slave_intr: %02" PRIx32, intr_mask);
```